# Mini Business Case Study - Sales Analytics for EcoTrek Solutions

## Objective: You are a new hire at EcoTrek Solutions, a U.S.-based company that specializes in eco-friendly travel accessories. As a business data analyst, your role involves analyzing sales trends and providing insights to help shape the company's marketing and production strategies.

In [109]:
# Import neccessary packages

import numpy as np
import pandas as pd
import scipy.stats
from scipy.stats import pearsonr

pd.set_option('display.max_columns', None)

In [110]:
# Create a DataFrame to read in data from provided file

df = pd.read_csv('Week_3_Temperature_DailySale.csv')

In [111]:
# Check the first few rows

df.head()

       Date  Daily Units Sold  Daily Temperature (C)
0  8/1/2025               380                     28
1  8/2/2025               420                     30
2  8/3/2025               350                     29
3  8/4/2025               300                     23
4  8/5/2025               340                     27

In [112]:
# Look at the data types

df.dtypes # Note that data is a string

Date                     object
Daily Units Sold          int64
Daily Temperature (C)     int64
dtype: object

In [113]:
# Look at summary staticstics of each field

df.describe().T

                       count    mean        std    min     25%    50%     75%  \
Daily Units Sold        20.0  345.50  34.561159  300.0  317.50  345.0  370.00   
Daily Temperature (C)   20.0   26.05   2.282081   22.0   24.75   26.0   27.25   

                         max  
Daily Units Sold       420.0  
Daily Temperature (C)   30.0  

In [114]:
# Rename columns to more user friendly names

df.rename(columns = {'Daily Units Sold': 'daily_units_sold',
                     'Daily Temperature (C)' : 'daily_temp'
                    },
          inplace = True
         )

In [115]:
# Conduct Pearson Correlation to determine relationship between daily sold units and daily temperature

r, p_value = pearsonr(df['daily_units_sold'], df['daily_temp'])
print(f"Pearson r: {r}, p-value: {p_value}") 

Pearson r: 0.8371391886262282, p-value: 4.171009629486727e-06


## The results indicate a strong positive relationship between daily sold units and daily temperature

In [116]:
# Sort the dataframe by daily units sold assuming descending

df.sort_values(by='daily_units_sold',ascending=False, inplace = True)

In [117]:
# Write the sorted dataframe in a new csv file

df.to_csv("GreenTote_analysis_Danut_George.csv", index=False)

## Since there is a strong positive correletion between daily sold units and daily temperature, I will reccomend Option 1 to my company to Develop a marketing campaign that considers weather information to strategize the sales of GreenTote

In [118]:
# Create a DataFrame to read in proced review file

reviews = pd.read_csv('Backpack_Reviews_with_Ratings.csv')
reviews.columns = reviews.columns.str.strip() # Clean the columns just in case
reviews.head() # review first few rows

                                              Review  Rating
0  I expected better quality for the price; not w...       1
1  It cleans easily after outdoor adventures; a b...       5
2   Handles are uncomfortable and dig into my hands.       1
3  Lightweight and easy to carry during hikes and...       5
4    The handles are sturdy and comfortable to hold.       5

In [119]:
# Import neccessary packages for SentimentIntensityAnalyzer

!pip install vaderSentiment
from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer

In [120]:
# Use SentimentIntensityAnalyze and for loops to loop through each review and give it a score and sentiment

# Initialize the Sentiment Intensity Analyzer
sia = SentimentIntensityAnalyzer()

# Empty list to hold sentiment labels
sentiments = []  

# Loop through each review
for review in reviews['Review']:
    score = sia.polarity_scores(str(review))['compound']
    if score >= 0.05:
        sentiments.append("positive")
    elif score <= -0.05:
        sentiments.append("negative")
    else:
        sentiments.append("neutral")

# Add new column to the same DataFrame
reviews['sentiment'] = sentiments

print(reviews)

                                                Review  Rating sentiment
0    I expected better quality for the price; not w...       1  positive
1    It cleans easily after outdoor adventures; a b...       5  positive
2     Handles are uncomfortable and dig into my hands.       1  negative
3    Lightweight and easy to carry during hikes and...       5  positive
4      The handles are sturdy and comfortable to hold.       5  positive
..                                                 ...     ...       ...
495  I appreciate the sustainable materials used; v...       5  positive
496  The eco-friendly claim seems questionable; no ...       1  negative
497  The eco-friendly claim seems questionable; no ...       1  negative
498  The eco-friendly claim seems questionable; no ...       1  negative
499  Great for carrying all my gear without feeling...       5  positive

[500 rows x 3 columns]


In [121]:
# Calculate the average rating og the reviews for each rating

avg_rating = reviews.groupby('sentiment')['Rating'].mean()

In [122]:
# Calculate the total number of reviews by sentiment
total_rating = reviews['sentiment'].value_counts()

In [123]:
# Display Results

print("Average Ratings by Sentiment:")
for sentiment, rating in avg_rating.items():
    print(f"{sentiment.capitalize():<8}: {rating:.2f}")

print("Total Amount of Ratngs by Sentiment:")
for sentiment, rating in total_rating.items():
    print(f"{sentiment.capitalize():<8}: {rating:.2f}")

Average Ratings by Sentiment:
Negative: 1.00
Neutral : 1.65
Positive: 4.69
Total Amount of Ratngs by Sentiment:
Positive: 247.00
Neutral : 135.00
Negative: 118.00


In [124]:
# Write the reviews dataframe in a new csv file

reviews.to_csv("Danut_George_sentiment.csv", index=False)

# Summary:
### Based on the results, nearly 50% of the reviews are positive which skews the optinion that most people liked the product. With the average positive rating being 4.69, nearly a perfect 5, it tells me that customers that enjoyed the product really rated it high. The averagre negative and neutral ratings tell me that those who left a negative or neutral review really did not enjoy the product as both ratings are very low. Overall, the products seems to be very polarizing - with about half really liking it and the other half really disliking it, not many people were truly in the middle in terms of rating.